pkgs/by-name/lp/LPCNet/package.nix

```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchurl, # to download test data
+  cmake,
+  codec2,
+  # for tests
+  octave,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "LPCNet";
+  version = "0.5";
+
+  src = fetchFromGitHub {
+    owner = "drowe67";
+    repo = "LPCNet";
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-9VxPD5GAIymr9TOAZmL69y+bcGAkdIJ6//EIeHDX9AQ=";
+    # Also download prebuilt neural network model that is needed during the build.
+    postFetch = ''
+      mkdir $out/build
+      curl \
```

Comment:
Wouldn't it be cleaner to use a separate `fetchurl` for the training data tarball (and link into the build dir)? That would decouple the code and the dataset.

Reply:
> Wouldn't it be cleaner to use a separate `fetchurl` for the training data tarball (and link into the build dir)? That would decouple the code and the dataset.

I was thinking about it, but I find it cleaner the fact that we have only 1 hash to maintain, and not 2... However choosing one or the other won't be significant in terms of version bumps to the package, as this tarball hasn't changed for a few years.

Reply:
curl should probably never be used directly in a package. The one hash solution means that the dataset also gets stored again when the package is updated. This would also make it easier to override the data set if necessary. 
I would recommend using `fetchurl`.

Reply:
> curl should probably never be used directly in a package. The one hash solution means that the dataset also gets stored again when the package is updated. This would also make it easier to override the data set if necessary. I would recommend using `fetchurl`.

Indeed being able to override it is a good reason to use `fetchurl` - in cases like the `rowetel.com` server is down.. Thanks :)